Tests/Realm.Tests/Database/InstanceTests.cs

```diff
@@ -141,7 +141,7 @@ public void DeleteRealmFailsIfOpenSameThread()
             using var openRealm = GetRealm();
 
             // Assert
-            Assert.Throws<RealmPermissionDeniedException>(() => Realm.DeleteRealm(RealmConfiguration.DefaultConfiguration));
+            Assert.Throws<RealmException>(() => Realm.DeleteRealm(RealmConfiguration.DefaultConfiguration));
```

Comment:
Is Core returning a non-generic exception that we can then wrap in `RealmPermissionDeniedException` or even better - something more concrete, such as `RealmInUseException`? You can see how exceptions are getting converted from Core to a .NET-compatible struct [here](https://github.com/realm/realm-dotnet/blob/fd1daa05d1bbc7e3169663cf9b541d16c6d8e542/wrappers/src/error_handling.cpp#L41) and then from an exception code to an actual managed exception [here](https://github.com/realm/realm-dotnet/blob/fd1daa05d1bbc7e3169663cf9b541d16c6d8e542/Realm/Realm/Exceptions/RealmException.cs#L48).

Reply:
Added with https://github.com/realm/realm-dotnet/pull/2401/commits/8b93783fba2a17735753fb541b6ffb26404f7ef9.